decidim-admin/app/models/decidim/admin/abilities/participatory_process_admin.rb

```diff
@@ -13,22 +13,32 @@ class ParticipatoryProcessAdmin
 
         def initialize(user)
           return if user.role?(:admin)
-          return unless ManageableParticipatoryProcessesForUser.for(user).any?
+          participatory_processes = ManageableParticipatoryProcessesForUser.for(user)
+          return unless participatory_processes.any?
 
           can :read, :admin_dashboard
 
           can :manage, ParticipatoryProcess do |process|
-            ManageableParticipatoryProcessesForUser.for(user).include?(process)
+            participatory_processes.include?(process)
           end
+
           cannot :create, ParticipatoryProcess
           cannot :destroy, ParticipatoryProcess
 
           can :manage, ParticipatoryProcessUserRole do |role|
             role.user != user
           end
 
-          can :manage, [ParticipatoryProcessStep, ParticipatoryProcessAttachment] do |step|
-            ManageableParticipatoryProcessesForUser.for(user).include?(step.participatory_process)
+          can :manage, ParticipatoryProcessAttachment do |step|
```

Comment:
I prefer for it to grow vertically rather than horizontally. Also, this way we reduce merge issues (as merges happen in different lines).